### PR TITLE
Add eld-code-reviewer skill

### PR DESCRIPTION
**Pull request by parth-07**
This commit adds eld-code-reviewer skill that can be used by any agentic coding harness (such as codex, claude, and opencode) for a thorough review of any eld PR.

### CODE REVIEW

**Review by Steven6798 — APPROVED** [review]

**Review by quic-seaswara — APPROVED** [review]

**Review by quic-areg — APPROVED** [review]
LGTM

File: skills/eld-code-reviewer/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+name: eld-code-reviewer
+description: >
+  Use for code reviews, PR reviews, patches, and diffs in the eld repository.
+  Prefer over generic review skills for ELD linker changes.
+---
+
+# ELD code review skill
+
+Use this skill for **every** review in the ELD repository — even when the user just
+says "review PR N" or "review this." It supersedes the generic `review`,
```
Comment (quic-shaksuma) [inline skills/eld-code-reviewer/SKILL.md:11]: Agent reads body text is only after trigger. Trigger rules (e.g. superseding generic review) should go into description so the skill activates correctly across agents

File: skills/eld-code-reviewer/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Review as a senior linker/toolchain engineer. Focus on linker correctness first.
+Avoid generic style comments unless they affect correctness, maintainability, diagnostics
+or readability. Along with this, focus on finding corner cases that are missed by the patch.
+
+If you have any questions, then ask instead of assuming things.
+
+# Important key information required to review the patch and build/test changes
+
+Use the gh skill to view the PR and its changes and comments.
```
Comment (quic-shaksuma) [inline skills/eld-code-reviewer/SKILL.md:23]: gh skill is a dependency and can break review workflow in some agents

File: skills/eld-code-reviewer/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+When reviewing PRs, always take full context: PR description, code change diff, and
+the conversation in PR comments.
+
+## Review priorities
+
+Prioritize:
```
Comment (quic-shaksuma) [inline skills/eld-code-reviewer/SKILL.md:30]: Priorities -> Priorities in Order (if strict order is intended)
Across agents, this can cause inconsistent workflows, its always better to specify an explicit flow/order for priorities

File: skills/eld-code-reviewer/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+Check reproduce functionality, map-file output, and diagnostics when the patch could affect them.
+Specify a clear "not affected" note when they are not affected.
+
+
+## Review philosophy
+
+Do not approve a patch merely because it compiles, existing tests pass, the code
+looks cleaner, or the patch is small.
+
+A good review comment should answer at least one of:
```
Comment (quic-shaksuma) [inline skills/eld-code-reviewer/SKILL.md:82]: These should have an evidence gate/findings confidence level.
These can be enhanced to produce a minimal evidence test that can actually be added to the PR later for better validation/coverage

File: skills/eld-code-reviewer/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- Is the code changes doing what the PR description / commit message saying?
+- Can this change introduce any non-determinism in the link / generated image?
+- Does this change require any change to reproduce functionality?
+- Would a map-file improvement be helpful with this change?
+- Can the solution be implemented in a better and more clean way?
+- Does each new function name clearly describe the condition it checks or action
+  it performs?
+- Does each new/modified function name clearly describe its side-effects?
+- A particularly bad coding/C++ practice is used.
+- There's a ABI-breaking change in the plugin framework interface (LinkerWrapper, and PluginADT)
```
Comment (quic-shaksuma) [inline skills/eld-code-reviewer/SKILL.md:102]: Split relatively big context topics for more accurate model response
